flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -1293,10 +1292,10 @@ private void checkpointStreamOperator(StreamOperator<?> op) throws Exception {
 			StreamEdge edge,
 			int outputIndex,
 			Environment environment,
-			String taskName,
```

Comment:
Why do we need this change?

Reply:
There are two options:

- Not pass the heavy-height `Environment` here and pass the individual required fields from it. 

- Pass the big component `Environment` instead of every tiny fields.

I prefer the first option but it might bring more refactoring works, so I take the second way here. The task name could be got directly from `Environment`, no need to explicitly pass it.